hypex/comparators/abstract.py

```diff
@@ -125,41 +130,224 @@ def _extract_dataset(
+            baseline_field (Optional[str], optional): The column to use as the baseline for comparison. Required if `compare_by` is 'columns' or 'columns_in_groups'. Defaults to None.
+
+        Returns:
+            Tuple: A tuple containing the baseline data and the compared data.
+
+        Raises:
+            NoRequiredArgumentError: If `baseline_field` is None and `compare_by` is 'columns' or 'columns_in_groups' or 'cross'.
+            ValueError: If `compare_by` is not one of the allowed values.
+        """
+
+        target_fields = Adapter.to_list(target_fields)
+        if compare_by == "groups":
+            baseline_data, compared_data = cls._split_for_groups_mode(
+                data, group_field, target_fields
+            )
+        elif compare_by == "columns":
+            baseline_data, compared_data = cls._split_for_columns_mode(
+                data, baseline_field, target_fields
+            )
+        elif compare_by == "columns_in_groups":
+            baseline_data, compared_data = cls._split_for_columns_in_groups_mode(
+                data, group_field, baseline_field, target_fields
+            )
+        elif compare_by == "cross":
+            baseline_data, compared_data = cls._split_for_cross_mode(
+                data, group_field, baseline_field, target_fields
+            )
+        else:
+            raise ValueError(
+                f"Wrong compare_by argument passed {compare_by}. It can be only one of the following modes: 'groups', 'columns', 'columns_in_groups', 'cross'."
+            )
+        return baseline_data, compared_data
+
+    @classmethod
+    def _split_for_groups_mode(
+        cls, data: Dataset, group_field: str, target_fields: List[str]
+    ):
+        if isinstance(target_fields, List) and len(target_fields) > 1:
+            raise ValueError(
+                "Only one Target field can be passed when the comparison is done by groups"
+            )
+        if not isinstance(group_field, str):
+            raise TypeError(
+                f"group_field must be one string, {type(group_field)} passed."
+            )
+
+        data_buckets = sorted(data.groupby(by=group_field, fields_list=target_fields))
+        baseline_data = cls._split_ds_into_columns([data_buckets.pop(0)])
+        compared_data = cls._split_ds_into_columns(data=data_buckets)
+        return baseline_data, compared_data
+
+    @classmethod
+    def _split_for_columns_mode(
+        cls, data: Dataset, baseline_field: str, target_fields: List[str]
+    ):
+
+        baseline_data = [(f"{baseline_field}", data[baseline_field])]
+        compared_data = [(f"{column}", data[column]) for column in target_fields]
+        return baseline_data, compared_data
+
+    @classmethod
+    def _split_for_columns_in_groups_mode(
+        cls,
+        data: Dataset,
+        group_field: str,
+        baseline_field: str,
+        target_fields: List[str],
+    ):
+        if baseline_field is None:
```

Comment:
I think you should use this checks outside of function or and into typing None, because you nmay have problems with mypy